src/abstractdataframe/abstractdataframe.jl

```diff
@@ -1534,16 +1548,30 @@ function fillcombinations(df::AbstractDataFrame, indexcols;
     return out_df
 end
 
+"""
+MergeDuplicates = Union{Nothing,Function}
+
+Wherever the `mergeduplicates` keyword argument is available it is either `nothing` or
+a `Function` that will be executed to combine duplicated columns (when `makeunique=false`)
+"""
+MergeDuplicates = Union{Nothing,Function}
+
 """
     hcat(df::AbstractDataFrame...;
-         makeunique::Bool=false, copycols::Bool=true)
+         makeunique::Bool=false, mergeduplicates::MergeDuplicates=nothing, copycols::Bool=true)
 
 Horizontally concatenate data frames.
 
 If `makeunique=false` (the default) column names of passed objects must be unique.
 If `makeunique=true` then duplicate column names will be suffixed
 with `_i` (`i` starting at 1 for the first duplicate).
 
+If `makeunique=false` and `mergeduplicates` is a `Function` then duplicate columns 
+will be combined by invoking the function with all values from those columns.
+e.g. `mergeduplicates=coalesce` will use the first non-missing value. Since `hcat` and
+`hcat!` are performed recursively for more than two frames, this `mergeduplicates` 
+function will only combine two columns at a time.
```

Comment:
it is not clear what happens if `makeuniqe=true` and `mergeduplicates is `Function`.